TextArea in TabView cannot be column encoded

boutinb opened the pull request:
Fixes https://github.com/jasp-stats/INTERNAL-jasp/issues/1745

The RCode Text Area should always generated an option with the 'isRCode' meta.
If a suboption has a meta (isRCode, shouldEncode, ...), this meta should be set also has a submeta.

JorisGoosen commented on Desktop/analysis/analysis.cpp:
```diff
@@ -431,59 +434,66 @@ Json::Value& Analysis::_getParentBoundValue(const QVector<JASPControl::ParentKey
-	Json::Value &	parentBoundValue	= _getParentBoundValue(parentKeys, found, true),
-					copyPBV				= parentBoundValue; 
+	Json::Value* parentBoundValue	= &_boundValues;
+	QVector<std::string> parents	= _getParentBoundValue(parentKeys, parentBoundValue, found, true);
```
Isnt it actually more sensible to give the child-values to the parent and let the parent add them?

Instead of letting the child get a weird ref to a Json::Value in its parent and then setting stuff on there?

JorisGoosen commented on Desktop/analysis/analysis.cpp:
```diff
@@ -431,59 +434,66 @@ Json::Value& Analysis::_getParentBoundValue(const QVector<JASPControl::ParentKey
 		if ((meta.isObject() || meta.isArray()) && meta.size() > 0)
-			_boundValues[".meta"][name] = meta;
+		{
+			Json::Value* metaBoundValue = &_boundValues[".meta"];
```
This block of code for instance  looks way too complicated. While in fact you just want to pass the `meta` to the parent right?

boutinb replied:
The main issue here is that I need a reference where the meta has to be set, but this reference depends on the parents vector, and in C++, you cannot reassign a reference to another object. That's why I need a pointer.

JorisGoosen reviewed (changes requested):
Well, ive requested some changes because I feel like the flow of information is the wrong way round...

I think instead of getting references to Json::Value somewhere in the parent we should be sending stuff to the parent instead and letting it handle putting stuff in it's boundvalues.

Maybe you have really good reasons for this though, in that case we can talk about tomorrow? (also in case you just want to talk about it ;) )

JorisGoosen commented:
Do you want me to review it again?

JorisGoosen commented:
And this should go into 0.16.3 ? 

Because in that case we need to redirect it to `stable`

boutinb committed:
TextArea in TabView cannot be column encoded

Fixes https://github.com/jasp-stats/INTERNAL-jasp/issues/1745

The RCode Text Area should always generated an option with the 'isRCode' meta.
If a suboption has a meta (isRCode, shouldEncode, ...), this meta should be set also has a submeta.

boutinb commented:
@JorisGoosen As https://github.com/jasp-stats/INTERNAL-jasp/issues/1568 is planned only for 0.17, I put this PR to development.
Can you review it again?

JorisGoosen commented on Desktop/analysis/analysis.cpp:
```diff
@@ -386,38 +386,40 @@ Json::Value Analysis::asJSON(bool withRSource) const
+			Json::Value& parentBoundValues = (*parentBoundValue)[parent.name];
 			if (!parentBoundValues.isNull() && parentBoundValues.isArray())
 			{
 				for (Json::Value & boundValue : parentBoundValues)
```
Should it not be `const Json::Value & boundValue`? 
Just to be sure you aren't changing other objects?

JorisGoosen commented on Desktop/analysis/analysis.cpp:
```diff
@@ -461,45 +464,53 @@ Json::Value& Analysis::_getParentBoundValue(const QVector<JASPControl::ParentKey
 
 	if (found && parentBoundValue.isObject())
 	{
 		parentBoundValue[name] = value;
```
Aaah ok I guess not `const` then otherwise this wouldnt work
